File: frontend/src/types/mood.ts
export interface MoodEntry {
  id: string;
  mood_score: number;
  mood_label: string | null;
  journal_text: string | null;
  tags: string[];
  recorded_at: string;
  created_at: string;
}

export interface MoodTrendPoint {
  date: string;
  score: number;
  label: string | null;
}

export interface MoodTrends {
  entries: MoodTrendPoint[];
  average: number;
  highest: number;
  lowest: number;
  total_entries: number;
}

export interface MoodStats {
  current_streak: number;
  longest_streak: number;
  total_checkins: number;
  average_score: number;
  most_common_label: string | null;
  monthly_summary: MoodTrendPoint[];
}

export interface MoodCalendarEntry {
  date: string;
  score: number;
  label: string | null;
}

export const MOOD_EMOJIS: Record<number, string> = {
  1: "😭",
  2: "😢",
  3: "😔",
  4: "😕",
  5: "😐",
  6: "🙂",
  7: "😊",
  8: "😄",
  9: "🥰",
  10: "🌟",
};

export const MOOD_LABELS: Record<number, string> = {
  1: "非常低落",
  2: "低落",
  3: "不太好",
  4: "有点差",
  5: "一般",
  6: "还行",
  7: "不错",
  8: "挺好",
  9: "很好",
  10: "非常好",
};

File: backend/app/schemas/mood.py
from pydantic import BaseModel, Field


class MoodCheckinRequest(BaseModel):
    mood_score: int = Field(ge=1, le=10)
    mood_label: str | None = None
    journal_text: str | None = Field(default=None, max_length=2000)
    tags: list[str] = Field(default_factory=list)


class MoodEntryResponse(BaseModel):
    id: str
    mood_score: int
    mood_label: str | None
    journal_text: str | None
    tags: list[str]
    recorded_at: str
    created_at: str


class MoodTrendPoint(BaseModel):
    date: str
    score: float
    label: str | None


class MoodTrendsResponse(BaseModel):
    entries: list[MoodTrendPoint]
    average: float
    highest: float
    lowest: float
    total_entries: int


class MoodStatsResponse(BaseModel):
    current_streak: int
    longest_streak: int
    total_checkins: int
    average_score: float
    most_common_label: str | None
    monthly_summary: list[MoodTrendPoint]
